hardware/1Wire/1WireByKernel.cpp

```diff
@@ -551,7 +543,9 @@ void C1WireByKernel::GetDevice(const std::string& deviceName, /*out*/_t1WireDevi
 	device.family = ToFamily(deviceName.substr(0, 2));
 
 	// Device Id (6 chars after '.')
-	std_to_upper(deviceName.substr(3, 3 + 6 * 2), device.devid);
+	auto str = deviceName.substr(3, 3 + 6 * 2);
```

Comment:
Could this be declared as std::string

Reply:
auto deduces to std::string.

Reply:
Yep, and this way we can use auto everywhere.
I vote readability over optimization

Reply:
Could you revert this change and declare it as std::string ?